### PR TITLE
[#1892] improvement(docs): Provide guidance on configuring memory-related configs

### PR DESCRIPTION
**Pull request by rickyma**
### What changes were proposed in this pull request?

Provide guidance on configuring memory-related configs.

### Why are the changes needed?

For https://github.com/apache/incubator-uniffle/issues/1892.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No need.

### CODE REVIEW

File: docs/server_guide.md [hunk 1]
```diff
@@ -145,20 +145,46 @@ Finally, to improve the speed of writing to HDFS for a single partition, the val
...
+So, `rss-env.sh` will be:
+
+```
+XMX_SIZE=40g 
+MAX_DIRECT_MEMORY_SIZE=360g
+```
+
+##### server.conf
+
+Generally, `rss.server.read.buffer.capacity` of 20g is enough, you can pay more attention to the metric `read_used_buffer_size`.
```
Comment (jerqi) [inline docs/server_guide.md:178]: What's the percentage of the memory?

Comment (rickyma) [reply docs/server_guide.md:178]: We don't need that much read buffer. Currently, we recommend to set:
`rss.server.read.buffer.capacity` : `rss.server.buffer.capacity` = 1 : 18

which is approximately 20g : 330g in this example.

Comment (rickyma) [reply docs/server_guide.md:179]: I have added into the doc.

**Review by jerqi — APPROVED** [review]
LGTM.

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Comment by rickyma** [comment]
Merged to master. Thanks for your reviews @jerqi @xianjingfeng

**Pull request merged**